lib/components/form/call-taker/date-time-picker.tsx

```diff
@@ -121,69 +151,32 @@ const DateTimeOptions = ({
   )
   const [date, setDate] = useState<string | undefined>(initialDate)
   const [time, setTime] = useState<string | undefined>(initialTime)
-  const [typedTime, setTypedTime] = useState<string | undefined>(initialTime)
+  const [typedTime, setTypedTime] = useState<string | undefined>(
+    safeFormat(parseInputAsTime(homeTimezone, time, date), timeFormat, {
+      timeZone: homeTimezone
+    })
+  )
 
   const timeRef = useRef(null)
 
   const intl = useIntl()
 
-  /**
-   * Parse a time input expressed in the agency time zone.
-   * @returns A date if the parsing succeeded, or null.
-   */
-  const parseInputAsTime = (
-    timeInput: string = getCurrentTime(homeTimezone),
-    date: string = getCurrentDate(homeTimezone)
-  ) => {
-    if (!timeInput) timeInput = getCurrentTime(homeTimezone)
-
-    // Match one of the supported time formats
-    const matchedTimeFormat = SUPPORTED_TIME_FORMATS.find((timeFormat) =>
-      isMatch(timeInput, timeFormat)
-    )
-    if (matchedTimeFormat) {
-      const resolvedDateTime = format(
-        parse(timeInput, matchedTimeFormat, new Date()),
-        'HH:mm:ss'
-      )
-      return toDate(`${date}T${resolvedDateTime}`)
-    }
-    return ''
-  }
-
-  const dateTime = parseInputAsTime(time, date)
+  const dateTime = parseInputAsTime(homeTimezone, time, date)
 
   // Update state when external state is updated
   useEffect(() => {
     if (initialDate !== date) setDate(initialDate)
     if (initialTime !== time) {
-      setTime(initialTime)
+      handleTimeChange(initialTime || '')
     }
   }, [initialTime, initialDate])
```

Comment:
eslint is highlighting missing dependencies here: `date`, `time`, and `handleTimeChange`, could you add them to the dependency array as well.

Reply:
the problem is these useEffect hooks are propagating updates from the store into the component, adding those deps would cause a loop. Otherwise I would have just replaced them with callbacks

Reply:
There is one in #[caf809c](https://github.com/opentripplanner/otp-react-redux/pull/1316/commits/caf809ca655e5b24f7c9705ae66de21a3b6426bb)